src/transformers/tokenization_utils.py

```diff
@@ -507,18 +612,39 @@ def vocab_size(self) -> int:
     def is_fast(self):
         return False
 
+    @property
+    def max_len(self):
+        """ Kept here for backward compatibility.
+            Now renamed to `model_max_length` to avoid ambiguity.
+        """
+        return self.model_max_length
+
+    @property
+    def max_len_single_sentence(self):
+        return self.model_max_length - self.num_special_tokens_to_add(pair=False)
+
+    @property
+    def max_len_sentences_pair(self):
+        return self.model_max_length - self.num_special_tokens_to_add(pair=True)
```

Comment:
Removing these from models and moving them as simple properties

Reply:
Cool, this is cleaner.